File: day01/day01.kt
import kotlin.io.path.Path
import kotlin.io.path.readLines
import kotlin.math.absoluteValue

val whitespace = "\\s+".toRegex()

fun main() {
    val part1 = Path("input.txt")
        .readLines()
        .map { it.split(whitespace).map { it.toInt() } }
        .let { p -> p.map { it[0] }.sorted() zip p.map { it[1] }.sorted() }
        .map { (it.first - it.second).absoluteValue }
        .sum()
    println(part1)
}

File: day01/day01.py
with open("input.txt", "r") as input_file:
    lines = [tuple(map(int, line.strip().split())) for line in input_file]

first_numbers = sorted(line[0] for line in lines)
second_numbers = sorted(line[1] for line in lines)
lines = [(min(line), max(line)) for line in zip(first_numbers, second_numbers)]

sum_distances = sum(abs(line[0] - line[1]) for line in lines)
print(sum_distances)

similarity_scores = [n * second_numbers.count(n) for n in first_numbers]
print(sum(similarity_scores))
